### ISSUE TITLE
Data regime

### ISSUE THREAD

**Issue by klh5** [alan-turing-institute/ARC-Speech-Synthesis#19]
Decide more formally on how to curate the current dataset for consistency across models.

**Comment by alexeatscake** [comment]
I already have a little bit of code that might be relevant for this. 
It produces this Pandas DataFrame.
Would you want me to put it in a branch?

<img width="639" alt="Screenshot 2024-04-18 at 09 37 34" src="https://github.com/alan-turing-institute/ARC-Speech-Synthesis/assets/67908629/5bd52383-3223-4b0e-9d57-5aa0a053664a">

**Comment by klh5** [comment]
- [ ] Check whether the list of samples used to train StarGANv2 is actually known

**Comment by alexeatscake** [comment]
**Chat on 2024-04-22**

We want to pick a set of samples so that: 

* The dataset is the same between each of the models so we are sure we are separating the models and not by speaker features
* The source speaker clips are not repeated so we are sure that the model is learning features from the speaker's voice and not the choice of words.
* Speakers from the StarGAN-VC2 training data for targets so we can make use of this pre-trained model.
* Not sentences from the rainbow phrases as the choice of words being the same across speakers probably isn't representative of most use cases.
* Clips that are longer than two seconds long as the very short clips might cause outliners.

We will use 10 targets and 10 sources. We could extend this up to 20 targets and still stay in the StarGAN-VC2 training set. For sources, we will have many possibilities and we will choose ones all outside the StarGAN-VC2 training set.

For the target speakers, we will pick 5 clips per speaker giving 50 in total.  
For the source speakers, we will pick 100 per speaker giving 1000 in total.  

**For models that style encoding is needed**  
For each target speaker clip will use two clips from each source speaker. This means every target speaker and source speaker combination gets 10 samples generated. These clips will be paired by index so that it is reproducible between the three models.

**For models that require pretraining**  
For each target speaker, a model is trained using all the clips in the trainset from the HuggingFace VCTK dataset. Then 100 source clips are passed through each pretrained model, making sure they match the pairing used in the style encoding model.

These processes generate 1000 audio clips per model per training. As the model data is identical and the source speaker data never repeats a sample as long as the same test train split is used for each model there shouldn't be any systematic issue with training a classifier.

**Comment by klh5** [comment]
As far as I can tell, it's not possible to determine exactly which clips went into the StarGANv2 training set. As a preparation step the audio clips are arranged into fixed-length chunks, and there is a list of which chunks are used for train/val in the [repo](https://github.com/yl4579/StarGANv2-VC/tree/main/Data), but no way to know which clips are in which chunks as the split is random.

I'd suggest that where possible we just use the train/val/test chunks as they appear in the StarGANv2 repo for fine-tuning. I will also create a new split for cases where these chunks aren't suitable. While I would have preferred to use the same data for all models, the primary concern was that StarGAN would have an advantage if sentences appeared at inference time that had already been seen during training. This shouldn't happen because we'll be using source speakers that definitely weren't part of the training set, and avoiding any utterances that are duplicated across speakers (e.g. the rainbow phrases).
